File: api/src/main/java/put/poznan/sport/entity/Coach.java
package put.poznan.sport.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;
import put.poznan.sport.entity.sportFacility.SportFacility;

import java.util.List;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Builder
public class Coach {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String surname;

    @Column(nullable = false)
    private String imageUrl;

    @ManyToOne
    @JoinColumn(name = "sportFacilitiesId", nullable = false)
    @JsonIgnore
    private SportFacility sportFacility;

    @OneToMany(mappedBy = "coach", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    @JsonIgnore
    private List<TrainingSession> trainingSessions;
}

File: mobile_app/lib/models/coach/coach.dart
class Coach {
  int? id;
  String? name;
  String? surname;
  int? sportFacilitiesId;
  String? imageUrl;

  Coach({
    this.id,
    this.name,
    this.surname,
    this.sportFacilitiesId,
    this.imageUrl,
  });

  factory Coach.fromJson(Map<String, dynamic> json) {
    return Coach(
      id: json['id'],
      name: json['name'],
      surname: json['surname'],
      sportFacilitiesId: json['sportFacilitiesId'],
      imageUrl: json['imageUrl'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'surname': surname,
      'sportFacilitiesId': sportFacilitiesId,
      'imageUrl': imageUrl,
    };
  }
}
